src/components/structures/MessagePanel.tsx

```diff
@@ -743,7 +745,7 @@ export default class MessagePanel extends React.Component<IProps, IState> {
                     enableFlair={this.props.enableFlair}
                     showReadReceipts={this.props.showReadReceipts}
                     callEventGrouper={callEventGrouper}
-                    hideSender={this.props.room?.getMembers().length <= 2 && this.props.layout === Layout.Bubble}
+                    hideSender={isDirectMessage && this.props.layout === Layout.Bubble}
```

Comment:
Maybe this can also be short-circuited by swapping the ordering of the checks, it'd be easier to check if it's a bubble layout, than to calculate those members every event.

Reply:
I did have that thought too, though I also preferred hoisting it up to be a prop itself, so that its resolved only once per event/tile rather than once for each